File: 36-running-median/rust/src/main.rs
use serde::Deserialize;
use std::io::{self, Read};

#[derive(Deserialize)]
struct Input {
    stream: Vec<i64>,
}

fn solve(stream: &[i64]) -> Vec<String> {
    // TODO: return median after each insertion
    // Format: integer if count is odd, one-decimal float if even (e.g. "5" or "4.5")
    let _ = stream;
    todo\!()
}

fn main() {
    let mut buf = String::new();
    io::stdin().read_to_string(&mut buf).unwrap();
    let inp: Input = serde_json::from_str(&buf).unwrap();
    println\!("{}", solve(&inp.stream).join(" "));
}

File: 36-running-median/python/main.py
import json
import sys


def solve(stream):
    # TODO: return median after each insertion as a list of formatted strings
    # Format: "5" (odd count) or "4.5" (even count, one decimal)
    pass


def main():
    obj = json.load(sys.stdin)
    print(*solve(obj["stream"]))


if __name__ == "__main__":
    main()
